luxonis_ml/data/augmentations/augmentations.py

```diff
@@ -0,0 +1,359 @@
+        """Prepare labels to be compatible with albumentations.
+
+        @type labels: Dict[str, np.ndarray]
+        @param labels: Dict with labels
+        @type height: int
+        @param height: Input image height
+        @type width: int
+        @param width: Input image width
+        @type n_keypoints: int
+        @param n_keypoints: Number of keypoints per instance
+        @type return_mask: bool
+        @param return_mask: Whether to compute and return mask
+        @rtype: Tuple[np.ndarray, Optional[np.ndarray], np.ndarray,
+            np.ndarray, np.ndarray, np.ndarray, np.ndarray]
+        @return: Labels in albumentations format
+        """
+
+        mask = None
+        if return_mask:
+            mask = prepare_mask(labels, height, width)
+
+        bboxes, bboxes_classes = prepare_bboxes(labels)
+        keypoints_points, keypoints_visibility, keypoints_classes = (
+            prepare_keypoints(labels, height, width, n_keypoints)
+        )
+
+        data = {
+            "bboxes": bboxes,
+            "bboxes_classes": bboxes_classes,
+            "keypoints": keypoints_points,
+            "keypoints_visibility": keypoints_visibility,
+            "keypoints_classes": keypoints_classes,
+        }
+        if mask is not None:
+            data["mask"] = mask
+
+        return data
+
+    def _post_transform(
+        self,
+        data: Dict[str, np.ndarray],
+        metadata: Dict[str, np.ndarray],
+        n_segmentation_classes: int,
+        n_keypoints: int,
+        present_labels: Set[str],
+    ) -> Tuple[
+        np.ndarray,
+        Dict[str, np.ndarray],
+    ]:
+        """Postprocessing of albumentations output to LuxonisLoader
+        format.
+
+        @type data: Dict[str, np.ndarray]
+        @param data: Output data from albumentations
+        @type metadata: Dict[str, np.ndarray]
+        @param metadata: Metadata
+        @type n_segmentation_classes: int
+        @param n_segmentation_classes: Number of segmentation classes
+        @type n_keypoints: int
+        @param n_keypoints: Number of keypoints per instance
+        @type present_labels: Set[str]
+        @param present_labels: Set of present labels
+        @rtype: Tuple[np.ndarray, Dict[str, np.ndarray]]
+        @return: Image and labels
+        """
+
+        out_image = data["image"]
+        image_height, image_width, _ = out_image.shape
+        out_image = out_image.astype(np.float32)
+
+        out_mask = None
+        if "mask" in data:
+            out_mask = post_process_mask(data["mask"], n_segmentation_classes)
+
+        out_bboxes = post_process_bboxes(
+            data["bboxes"], np.expand_dims(data["bboxes_classes"], axis=-1)
+        )
+        out_keypoints = post_process_keypoints(
+            data["keypoints"],
+            data["keypoints_visibility"],
+            data["keypoints_classes"],
+            n_keypoints,
+            image_height,
+            image_width,
+        )
```

Comment:
Are invisible keypoints removed, or are their values just set to 0?

Reply:
Values are set to 0 as per coco convention.

Reply:
So zeros for invisible keypoints present in the targets during training?

Reply:
Yeah. The model predicts fixed number of keypoints so we can't remove the invisible ones. Instead, they are set to 0.

Reply:
I see, thanks!